scripts/py_matter_idl/matter_idl/backwards_compatibility.py

```diff
@@ -0,0 +1,291 @@
+            return
+
+        if original.base_type != updated.base_type:
+            self._mark_incompatible(
+                f"Bitmap {cluster_name}::{original.name} switched base type from {original.base_type} to {updated.base_type}")
+
+        # Validate that all old entries exist
+        for entry in original.entries:
+            # old entry must exist and have identical code
+            existing = [item for item in updated.entries if item.name == entry.name]
+            if len(existing) == 0:
+                self._mark_incompatible(f"Bitmap {original.name} removed entry {entry.name}")
+            elif existing[0].code != entry.code:
+                self._mark_incompatible(
+                    f"Bitmap {original.name} changed code for entry {entry.name} from {entry.code} to {existing[0].code}")
+
+    def _check_event_compatible(self, cluster_name: str, event: Event, updated_event: Optional[Event]):
+        if not updated_event:
+            self._mark_incompatible(f"Event {cluster_name}::{event.name} was removed")
+            return
+
+        if event.code != updated_event.code:
+            self._mark_incompatible(f"Event {cluster_name}::{event.name} code changed from {event.code} to {updated_event.code}")
+
+        self._check_field_lists_are_the_same(f"Event {cluster_name}::{event.name}", event.fields, updated_event.fields)
+
+    def _check_command_compatible(self, cluster_name: str, command: Command, updated_command: Optional[Command]):
+        self.logger.debug(f"  Checking command {cluster_name}::{command.name}")
+        if not updated_command:
+            self._mark_incompatible(f"Command {cluster_name}::{command.name} was removed")
+            return
+
+        if command.code != updated_command.code:
+            self._mark_incompatible(
+                f"Command {cluster_name}::{command.name} code changed from {command.code} to {updated_command.code}")
+
+        if command.input_param != updated_command.input_param:
+            self._mark_incompatible(
+                f"Command {cluster_name}::{command.name} input changed from {command.input_param} to {updated_command.input_param}")
+
+        if command.output_param != updated_command.output_param:
+            self._mark_incompatible(
+                f"Command {cluster_name}::{command.name} output changed from {command.output_param} to {updated_command.output_param}")
+
+        if command.qualities != updated_command.qualities:
+            self._mark_incompatible(
+                f"Command {cluster_name}::{command.name} qualities changed from {command.qualities} to {updated_command.qualities}")
+
+    def _check_struct_compatible(self, cluster_name: str, original: Struct, updated: Optional[Struct]):
+        self.logger.debug(f"  Checking struct {original.name}")
+        if not updated:
+            self._mark_incompatible(f"Struct {cluster_name}::{original.name} has been deleted.")
```

Comment:
So if a struct gets renamed this will mark things as incompatible?

Reply:
Yes, the intent was to mostly "flag this for human review". If some things are too often flagged we can update the rules to become smarter.

Ideally this would be more strict than needed rather than the reverse.